test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -796,4 +1225,270 @@ class QueryTests {
+    // Class instantiation with property setting helpers
+    // --------------------------------------------------
+
+    private fun <C> getInstance(
+        propertyDescriptor: PropertyDescriptor,
+        instance: C,
+        index: Int? = null
+    ): C {
+        return when (propertyDescriptor.property.returnType.classifier as KClass<*>) {
+            Int::class -> setProperty(
+                instance,
+                propertyDescriptor.property as KMutableProperty1<C, Int?>,
+                propertyDescriptor.values as List<Int?>,
+                index
+            )
+            Long::class -> setProperty(
+                instance,
+                propertyDescriptor.property as KMutableProperty1<C, Long?>,
+                propertyDescriptor.values as List<Long?>,
+                index
+            )
+            Short::class -> setProperty(
+                instance,
+                propertyDescriptor.property as KMutableProperty1<C, Short?>,
+                propertyDescriptor.values as List<Short?>,
+                index
+            )
+            Double::class -> setProperty(
+                instance,
+                propertyDescriptor.property as KMutableProperty1<C, Double?>,
+                propertyDescriptor.values as List<Double?>,
+                index
+            )
+            Float::class -> setProperty(
+                instance,
+                propertyDescriptor.property as KMutableProperty1<C, Float?>,
+                propertyDescriptor.values as List<Float?>,
+                index
+            )
+            RealmInstant::class -> setProperty(
+                instance,
+                propertyDescriptor.property as KMutableProperty1<C, RealmInstant?>,
+                propertyDescriptor.values as List<RealmInstant?>,
+                index
+            )
+            else -> throw IllegalArgumentException("Only numerical properties and timestamps are allowed.")
+        }
+    }
+
+    private fun <C, T> setProperty(
+        instance: C,
+        property: KMutableProperty1<C, T?>,
+        data: List<T?>,
+        index: Int? = null
+    ): C = instance.apply {
+        when (index) {
+            null -> property.set(instance, null)
+            else -> property.set(instance, data[index])
+        }
+    }
+
+    // -------------------------------------------------
+    // Aggregator helpers used to initialize structures
+    // -------------------------------------------------
+
+    private fun expectedSum(clazz: KClass<*>, kType: KType): Any =
+        expectedAggregator(clazz, kType, AggregatorQueryType.SUM)
+
+    private fun expectedMax(clazz: KClass<*>, kType: KType): Any =
+        expectedAggregator(clazz, kType, AggregatorQueryType.MAX)
+
+    private fun expectedMin(clazz: KClass<*>, kType: KType): Any =
+        expectedAggregator(clazz, kType, AggregatorQueryType.MIN)
+
+    @Suppress("LongMethod", "ComplexMethod")
+    private fun expectedAggregator(
```

Comment:
Seems very bloated. Can't you just assign an aggregator to a variable by referencing `::minOrNull()`, `::maxOrNull`, etc. and then apply it on a set looked up from a map of `Map<Pair<KClass<*>, Boolean>. Collection<*>`. Pretty hard to catch errors here, because you loose the overview.

Reply:
Way better, but I still think that I would have done it with separate mappings for:
- _classifier/KClass_ to test collection, and 
- `AggregatorQueryType` to the Collection aggregator
And then apply the aggregator on the collection in the end. Maybe your proposal is better handling the special case that sum is not supported for RealmInstant

Reply:
Yes, it's a bummer that `sum` and `RealmInstant` break the general case. I will leave it like this for now though, we can revisit it once we put time aside to standardise our exhaustive testing strategy.